### PR TITLE
fix: identity resolution

### PR DESCRIPTION
**Pull request by chris13524**
# Breaking Changes

N/A

# Changes

- fix: identity API not using mainnet
- fix: identity API not being used for ethers SDKs

[Slack conversation](https://walletconnect.slack.com/archives/C03RVH94K5K/p1712418227203839?thread_ts=1712373328.103609&cid=C03RVH94K5K)

### CODE REVIEW

File: packages/ethers/src/client.ts [hunk 1]
```diff
@@ -881,20 +881,28 @@ export class Web3Modal extends Web3ModalScaffold {
...
-      if (name) {
-        this.setProfileName(name)
-      }
-      if (avatar) {
-        this.setProfileImage(avatar)
+    try {
+      const { name, avatar } = await this.fetchIdentity({
+        address
+      })
+      this.setProfileName(name)
```
Comment (glitch-txs) [inline packages/ethers/src/client.ts:888]: show we use conditionals here too? e.g. some address can have `name` but not `avatar`. Not sure how this is handled internally

Comment (chris13524) [reply packages/ethers/src/client.ts:888]: The API returns `null` if there is no avatar so the conditional is redundant?

**Review by glitch-txs — APPROVED** [review]

File: packages/ethers/src/client.ts [hunk 2]
```diff
@@ -881,20 +881,28 @@ export class Web3Modal extends Web3ModalScaffold {
...
+        if (name) {
+          this.setProfileName(name)
+        }
+        if (avatar) {
+          this.setProfileImage(avatar)
+        }
+      } else {
+        this.setProfileName(null)
+        this.setProfileImage(null)
       }
```
Comment (arein) [inline packages/ethers/src/client.ts:905]: can we just remove this code? It was for fallbacks if our API is down

**Review by arein — APPROVED** [review]

File: packages/ethers/src/client.ts (hunk 2 above)
Comment (chris13524) [reply packages/ethers/src/client.ts:905]: Isn't it nice to have the fallback still?

**Review by tomiir — APPROVED** [review]

File: packages/ethers/src/client.ts (hunk 2 above)
Comment (chris13524) [reply packages/ethers/src/client.ts:904]: I think it's not a bad thing to have. We also had downtime this week that this fallback handled.

**Pull request merged**